tasks/update_go.py

```diff
@@ -31,9 +31,6 @@ def _get_archive_extension(os: str) -> str:
 
 def _get_expected_sha256(version: str) -> List[Tuple[Platform, str]]:
     """returns a map from platform to sha of the archive"""
-    # weirdly, the stored sha256 for round versions don't have a ".0" in the version
-    # while the archives have the ".0" suffix
-    version = version.removesuffix(".0")
```

Comment:
Hum you are right it seems that this has changed since 1.21:
- 1.20 without trailing `.0`: https://storage.googleapis.com/golang/go1.20.linux-arm64.tar.gz.sha256
- 1.21 has trailing `.0`: https://storage.googleapis.com/golang/go1.21.0.linux-arm64.tar.gz.sha256